### PR TITLE
chore: split smash url per env

### PR DESCRIPTION
**Pull request by bernokl**
Goal is to let us update smash url  per environment

### CODE REVIEW

**Review by iccicci — CHANGES_REQUESTED** [review]
Many URLs lack the `/api/v1` path.

More than this there are some indentation problems.

File: nix/cardano-services/deployments/default.nix [hunk 1]
```diff
@@ -211,6 +212,7 @@ in
           blockfrost-worker.enabled = true;
           pg-boss-worker.enabled = true;
           pg-boss-worker.metadata-fetch-mode = "smash";
+          pg-boss-worker.smash-url = "https://preview-smash.world.dev.cardano.org";
```
Comment (iccicci) [inline nix/cardano-services/deployments/default.nix:215]: I would say path `/api/v1` is required here as well.

Is there some reason I don't know or is it just a refuse which still needs to be addressed?

File: nix/cardano-services/deployments/default.nix [hunk 2]
```diff
@@ -401,8 +405,11 @@ in
           backend.allowedOrigins = lib.concatStringsSep "," allowedOriginsDev;
           backend.hostnames = ["${final.namespace}.${baseUrl}"];
 
-          blockfrost-worker.enabled = true;
           pg-boss-worker.enabled = true;
+          pg-boss-worker.metadata-fetch-mode = "smash";
+          pg-boss-worker.smash-url = "https://preprod-smash.world.dev.cardano.org/";
```
Comment (iccicci) [inline nix/cardano-services/deployments/default.nix:410]: Same comment about path here.

File: nix/cardano-services/deployments/default.nix [hunk 3]
```diff
@@ -476,6 +483,8 @@ in
 
           backend.hostnames = ["${final.namespace}.${baseUrl}"];
           blockfrost-worker.enabled = true;
+          pg-boss-worker.metadata-fetch-mode = "direct";
+          pg-boss-worker.smash-url = "https://preprod-smash.world.dev.cardano.org/";
```
Comment (iccicci) [inline nix/cardano-services/deployments/default.nix:487]: Same comment about path here.

File: nix/cardano-services/deployments/default.nix [hunk 4]
```diff
@@ -556,6 +565,8 @@ in
 
           backend.hostnames = ["backend.${final.namespace}.eks.${baseUrl}" "${final.namespace}.${baseUrl}"];
           blockfrost-worker.enabled = true;
+          pg-boss-worker.metadata-fetch-mode = "direct";
+          pg-boss-worker.smash-url = "https://preprod-smash.world.dev.cardano.org/";
```
Comment (iccicci) [inline nix/cardano-services/deployments/default.nix:569]: Same comment about path here.

File: nix/cardano-services/deployments/default.nix [hunk 5]
```diff
@@ -637,6 +648,8 @@ in
           backend.hostnames = ["backend.${final.namespace}.eks.${baseUrl}" "${final.namespace}.${baseUrl}"];
           blockfrost-worker.enabled = true;
           pg-boss-worker.enabled = true;
+          pg-boss-worker.metadata-fetch-mode = "direct";
+          pg-boss-worker.smash-url = "https://preview-smash.world.dev.cardano.org/";
```
Comment (iccicci) [inline nix/cardano-services/deployments/default.nix:652]: Same comment about path here.

File: nix/cardano-services/deployments/default.nix [hunk 6]
```diff
@@ -705,6 +718,8 @@ in
           region = "us-east-2";
 
           blockfrost-worker.enabled = false;
+          pg-boss-worker.metadata-fetch-mode = "direct";
+					pg-boss-worker.smash-url = "https://sanchonet-smash.world.dev.cardano.org/";
```
Comment (iccicci) [inline nix/cardano-services/deployments/default.nix:722]: Same comment about path here.

**Review by iccicci — CHANGES_REQUESTED** [review]
Sorry, I realized about another problem only now.

Since we are configuring all the environments to use smash servers, `pg-boss-worker.metadata-fetch-mode` should be set to `"smash"` for all the environments as well; maybe it worth to apply this setting as a global setting rather than having it on each environment.

File: nix/cardano-services/deployments/default.nix (hunk 1 above)
Comment (bernokl) [reply nix/cardano-services/deployments/default.nix:215]: I only did manual testing of the url provided in slack. Thank you for confirming it needs the /api/v1

**Comment by bernokl** [comment]
> Since we are configuring all the environments to use smash servers, pg-boss-worker.metadata-fetch-mode should be set to "smash" for all the environments as well; maybe it worth to apply this setting as a global setting rather than having it on each environment.


There is a default that used to be direct I set to smash and made the url a default as well so we can just override on deviation

**Comment by bernokl** [comment]
Manually confirmed code only made url update as expected when ran against dev-preview

**Review by iccicci — APPROVED** [review]
LGTM!

File: nix/cardano-services/deployments/default.nix [hunk 7]
```diff
@@ -113,12 +113,16 @@ in
...
-          resources.limits = mkPodResources "300Mi" "300m";
-          resources.requests = mkPodResources "150Mi" "200m";
+        pg-boss-worker =  let
+          smash-url = if values.network == "mainnet"
+                         then "https://smash.cardano-mainnet.iohk.io/api/v1"
+                         else "https://${values.network}-smash.world.dev.cardano.org/api/v1";
+        in {
+           enabled = false;
+           metadata-fetch-mode = "smash";
+           smash-url = smash-url;
```
Comment (gytis-ivaskevicius) [inline nix/cardano-services/deployments/default.nix:123]:
```suggestion
        pg-boss-worker =  {
           enabled = false;
           metadata-fetch-mode = "smash";
           smash-url = if values.network == "mainnet"
                         then "https://smash.cardano-mainnet.iohk.io/api/v1"
                         else "https://${values.network}-smash.world.dev.cardano.org/api/v1";```

**Review by gytis-ivaskevicius — APPROVED** [review]

**Review by michalrus — DISMISSED** [review]

**Review by gytis-ivaskevicius — APPROVED** [review]

**Review by michalrus — APPROVED** [review]

**Pull request merged**